### PR TITLE
bind_cols(<...>) -> tibble

### PR DESCRIPTION
**Pull request by romainfrancois**
``` r
library(dplyr, warn.conflicts = FALSE)

bind_cols(data.frame(x = 1), y = 2)
#>   x y
#> 1 1 2
bind_cols(tibble(x = 1), y = 2)
#> # A tibble: 1 x 2
#>       x     y
#> * <dbl> <dbl>
#> 1     1     2
bind_cols(x = 1, y = 2)
#> # A tibble: 1 x 2
#>       x     y
#>   <dbl> <dbl>
#> 1     1     2
```

<sup>Created on 2020-03-18 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

### CODE REVIEW

**Comment by romainfrancois** [comment]
Related to https://github.com/tidyverse/purrr/pull/752

**Comment by romainfrancois** [comment]
This should also fix `broom`: 

```
[master* ↓] 286.2 MiB ❯ revdepcheck::revdep_details(revdep = "broom")
══ Reverse dependency check ═════════════════════════════════════ broom 0.5.5 ══

Status: BROKEN

── Still failing

✖ checking package dependencies ... NOTE
✖ checking Rd cross-references ... NOTE

── Newly failing

✖ checking tests ...

── Before ──────────────────────────────────────────────────────────────────────
❯ checking package dependencies ... NOTE
  Packages suggested but not available for checking: 'btergm', 'xergm'

❯ checking Rd cross-references ... NOTE
  Unknown package ‘btergm’ in Rd xrefs

0 errors ✔ | 0 warnings ✔ | 2 notes ✖

── After ───────────────────────────────────────────────────────────────────────
❯ checking tests ...
  See below...

❯ checking package dependencies ... NOTE
  Packages suggested but not available for checking: 'btergm', 'xergm'

❯ checking Rd cross-references ... NOTE
  Unknown package ‘btergm’ in Rd xrefs

── Test failures ───────────────────────────────────────────────── test-all ────

> library(testthat)
> test_check("broom")
Loading required package: broom
── 1. Failure: tidy.muhaz (@test-muhaz.R#18)  ──────────────────────────────────
`output` inherits from `data.frame` not `tbl_df`.

── 2. Failure: glance.muhaz (@test-muhaz.R#24)  ────────────────────────────────
`output` inherits from `data.frame` not `tbl_df`.

══ testthat results  ═══════════════════════════════════════════════════════════
[ OK: 842 | SKIPPED: 3 | WARNINGS: 4 | FAILED: 2 ]
1. Failure: tidy.muhaz (@test-muhaz.R#18)
2. Failure: glance.muhaz (@test-muhaz.R#24)

Error: testthat unit tests failed
Execution halted

1 error ✖ | 0 warnings ✔ | 2 notes ✖
```

File: R/bind.r [hunk 1]
```diff
@@ -137,6 +137,9 @@ bind_cols <- function(...) {
   names(dots)[is_data_frame] <- ""
 
   out <- vec_cbind(!!!dots)
+  if (!any(map_lgl(dots, function(.x) is.data.frame(vec_ptype(.x))))) {
```
Comment (lionel-) [inline R/bind.r]: Is the `vec_ptype()` necessary here? it looks like this could be:

```suggestion
  if (!any(map_lgl(dots, is.data.frame))) {
```

Comment (romainfrancois) [reply R/bind.r]: oh yeah of course. Thanks.

**Pull request merged**